reactor-core/src/main/java/reactor/core/publisher/FluxHandleFuseable.java

```diff
@@ -231,14 +249,30 @@ public R poll() {
 					T v = s.poll();
 					R u;
 					if (v != null) {
-						handler.accept(v, this);
+						try {
+							handler.accept(v, this);
+						}
+						catch (Throwable error){
+							Throwable e_ = Operators.onNextPollError(v, error, actual.currentContext());
+							if (e_ != null) {
+								throw Exceptions.propagate(e_);
+							}
+							else {
+								done = false;
+								error = null;
```

Comment:
`this.error` is hidden by local `error` variable that you used here. need to replace with `this.error` or use a `reset()` method like you did so far